repos/system_upgrade/el7toel8/actors/storagescanner/libraries/storagescanner.py

```diff
@@ -75,65 +75,75 @@ def _get_partitions_info(partitions_path):
+
+            if len(entries) == 5:
+                entries.append('0')
+
+            if len(entries) != 6:
+                if any(value.startswith('#') for value in entries):
+                    remediation = (
+                        'Comments in the /etc/fstab file must be at the beginning of the line, your file has a'
+                        ' comment at the end of the line at line {}, please edit and fix this, for further'
+                        ' information read fstab man page (man 5 fstab).'.format(line)
+                    )
+                else:
+                    remediation = (
+                        'The /etc/fstab file must have at least 4 values and at most 6 per line, your file on the'
+                        ' line: {} has {} values, please edit and fix this, for further information read'
+                        ' fstab man page (man 5 fstab). '.format(line, len(entries))
                     )
-                    reporting.create_report([
-                        reporting.Title('Problems with parsing data in /etc/fstab'),
-                        reporting.Summary(summary),
-                        reporting.Severity(reporting.Severity.HIGH),
-                        reporting.Tags([reporting.Tags.FILESYSTEM]),
-                        reporting.Flags([reporting.Flags.INHIBITOR]),
-                        reporting.Remediation(hint=remediation),
-                        reporting.RelatedResource('file', '/etc/fstab')
-                    ])
-
-                    api.current_logger().error(summary)
-                    break
-
-                fs_spec, fs_file, fs_vfstype, fs_mntops, fs_freq, fs_passno = entries
-                yield FstabEntry(
-                    fs_spec=fs_spec,
-                    fs_file=fs_file,
-                    fs_vfstype=fs_vfstype,
-                    fs_mntops=fs_mntops,
-                    fs_freq=fs_freq,
-                    fs_passno=fs_passno)
+                summary = (
+                    'The fstab configuration file seems to be invalid. You need to fix it to be able to proceed'
+                    ' with the upgrade process.'
+                )
+                _add_error('Problems with parsing data in /etc/fstab', summary, remediation)
+
+                api.current_logger().error(summary)
+                continue
+
+            fs_spec, fs_file, fs_vfstype, fs_mntops, fs_freq, fs_passno = entries
+            # NOTE(ivasilev) As specified in OAMG-4107, before networking is working in the initramfs stage
+            # the upgrade should be inhibited if any CIFS entries are found.
+            if fs_vfstype == 'cifs':
+                summary = 'CIFS entries are not supported at the moment.'
+                _add_error('CIFS entry encountered in /etc/fstab', summary,
+                           'Please comment out the CIFS entry at line {}.'.format(line))
+                api.current_logger().error(summary)
+                continue
```

Comment:
The logic about evalutaion of data should not be here. This actor should provide only what's inside the fstab file. Next to CIFS, we need to inhibit even NFS, etc. The actual check for the CIFS paritions should be implemented in the separate actor. It can be analogy to `checknfs` or even maybe better, the `checknfs` actor should be extended.

Reply:
Of course, in case of extension of the NFS inhibitor, it would make sense to rename the actor.

Reply:
I'd rather have a different actor if that's ok.